components/api_version/src/lib.rs

```diff
@@ -311,10 +318,46 @@ mod tests {
                 Some(case.1),
                 case.2,
                 ApiVersion::V1ttl,
+                false,
             );
         }
         for case in &cases {
-            assert_raw_value_encode_decode_identity(case.0, Some(case.1), case.3, ApiVersion::V2);
+            assert_raw_value_encode_decode_identity(
+                case.0,
+                Some(case.1),
+                case.3,
+                ApiVersion::V2,
+                false,
+            );
+        }
+    }
+
+    #[test]
+    fn test_meta() {
+        // (user_value, expire_ts, ecoded_bytes_v2, is_delete)
+        let cases = vec![
+            // only deletion flag.
+            (&b""[..], None, &[2][..], true),
```

Comment:
Consider giving some of them a non-empty value.

Reply:
this case is to test logical delete scenario, so it just has one byte(ValueMeta). There are two cases with non-empty value in the case suit.